src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -168,8 +176,13 @@ public void finish() {
      *         output buffer
      */
     @Override
-    public int deflate(byte[] b, int off, int len ) {
-
+    public int deflate(byte[] b, int off, int len ) throws NullPointerException {
+        if(b == null) {
+            throw new NullPointerException("Input is null");
+        }
```

Comment:
This is covered in PR #122

Reply:
I can undo this changes if its covered in another PR